crates/primitives/utils/src/service.rs

```diff
@@ -1,21 +1,297 @@
 //! Service trait and combinators.
 
 use anyhow::Context;
-use std::panic;
+use futures::Future;
+use std::{fmt::Display, panic, sync::Arc};
 use tokio::task::JoinSet;
 
+#[repr(u8)]
+#[derive(Clone, Copy, PartialEq, Eq, Default, Debug)]
+pub enum MadaraCapability {
```

Comment:
i kinda dont like the name capability here
it's a just a running service right? why capability?

telemetry doesn't *provide* anything for example, i wouldn't call that a capability

Reply:
This has been fixed. Renamed this to `MadaraService` and made related changes across the codebase.